### ISSUE TITLE
Support for PHP 8

### ISSUE THREAD

**Issue by leewillis77** [weprovide/valet-plus#546]
- [x] I've checked the issue queue and could not find anything similar to my feature.
- [x] I'm on the latest version of valet-plus (`valet --version`): `Valet+ 2.0.0`

**Why implement this feature?**  
PHP 8 is due out at the end of November 2020. Will it be supported in Valet+ ?

**Comment by Neodork** [comment]
@leewillis77 Thanks for your feature request. I will pick this up. As for whether it will be reported, we'll definitely.

@henkrehorst Do we have 8.0 formula available already within our PHP automation?

**Comment by Neodork** [comment]
@leewillis77 I've talked to @henkrehorst and he confirmed that the automation can offer support for PHP 8.0 as of 26 November. So somewhere between November and early December we can offer support for 8.0.

**Comment by i6media** [comment]
Laravel already supports PHP 8.0: https://blog.laravel.com/laravel-php-8-support

Any time window for the release of Valet+?

**Comment by Neodork** [comment]
> So somewhere between November and early December we can offer support for 8.0.

Yes. See above. The formula is already supported: https://github.com/henkrehorst/homebrew-php/tree/master/Formula

**Comment by leewillis77** [comment]
There is a PHP 8.0 build available, but valet will not / can't use it:
```
$  brew install valet-php@8.0                               
Warning: henkrehorst/php/valet-php@8.0 8.0.0_11 is already installed and up-to-date
To reinstall 8.0.0_11, run `brew reinstall valet-php@8.0`
$ valet use 8.0
In PhpFpm.php line 156:                                                                                                    
  This version of PHP not available. The following versions are available: 5.6 7.0 7.1 7.2 7.3 7.4  
                                                                                                    

use [<service>] [<targetVersion>]
```

**Comment by lushkant** [comment]
Really hoping, v8 support gets added soon 🤞 🙏

**Comment by christopherholst** [comment]
We are also really hoping, PHP 8 support gets added soon!

**Comment by dphyled** [comment]
Yep, love Valet+ and hoping PHP 8 support will be added! 👍

**Comment by paulverdu** [comment]
Is there any information on when php 8.0 will be added to Valet Plus?

**Comment by mischabraam** [comment]
Basically PHP8.0 is in the 2.x branch (dev). I'm having an issue with geoip pecl dependency.

EDIT: https://blog.remirepo.net/post/2020/09/21/PHP-extensions-status-with-upcoming-PHP-8.0

**Comment by mischabraam** [comment]
https://github.com/weprovide/valet-plus/tree/feature/php-80

Feature branch above supports php 8.0. However, I had to update different dependencies and disable the geoip PECL extension. After that switching to php 7.2 or lower works... but then valet+ stops working. You can NOT switch back to 7.3 or newer, or do other commands. You'll have to do that manually with below commands after which valet+ will work fine again.
```
sudo brew services stop valet-php@<old>
sudo brew services start valet-php@<new>
brew link valet-php@<new> --force
```

So, I guess when this is released it will NOT be backwards compatible.

**Comment by christopherholst** [comment]
Is there already something new here, is it coming in a release soon?

**Comment by paulverdu** [comment]
> Is there already something new here, is it coming in a release soon?

I’ve just stopped using valet-plus and just install and configure everything on my own with Brew including PHP 8.

**Comment by i6media** [comment]
> Is there already something new here, is it coming in a release soon?

To be honest, the regular [Valet](https://laravel.com/docs/8.x/valet) is pretty Plus already these days. It already supports PHP 8 fully.

Switching PHP Versions, Magento & WordPress support is also included.

I've stopped using valet-plus as well, just like @paulverdu, and switched fully to Laravel Valet instead of plus.

**Comment by mischabraam** [comment]
I've disabled the platform check making my branch backwards compatible at least for php switching. Can anyone confirm this works?

**Comment by viktorbijlenga** [comment]
This issue probably can be closed since support for PHP 8.0 is released. 
For more info see Valet+ 2.2.0: https://github.com/weprovide/valet-plus/releases/tag/2.2.0
